pushapkscript/googleplay.py

```diff
@@ -10,23 +10,18 @@
 _EXPECTED_L10N_STRINGS_FILE_NAME = 'public/google_play_strings.json'
 
 
-def publish_to_googleplay(payload, product_config, apk_files, contact_google_play, google_play_strings_file=None):
-    track = payload['google_play_track']
-    valid_track_values = craft_valid_track_values(product_config.get('require_track'),
-                                                  product_config['has_nightly_track'])
-    if track not in valid_track_values:
```

Comment:
After our discussion last week about dynamically using new internal tracks (e.g.: putting Aurora on a temporary testing track), I started considering that a whitelist isn't quite the best fit. Looking at the pro-con, I found:

**Pros:**
* When an invalid track is specified (probably by typo?), we'll be warned at the beginning of the build, rather than relying on Google Play to validate the track

**Cons:**
* The end result of the whitelist is that it only protects us by forcing us to _write the track a second time_, which isn't particularly effective
* For temporary testing tracks, we have to clean up the whitelist after-the-fact
* Releng is the only team that manages `pushapkscript`, so I feel that this pulls control away from projects and forces changes to by handled by releng. We know that we want to keep all non-credential information within projects themselves - this is what declarative artifacts pushes for. I think that removing this `pushapkscript` whitelist is a step in same direction

Reply:
I agree a typo would definitely be caught by Google Play, and the temporary testing track workflow would be cumbersome.

Although, it may be worth protecting against bad valid tracks. For instance: if someone tries to push `org.mozilla.fennec_aurora` onto the production track. That said, this check has never existed, and we've never encountered that problem so far. As a consequence, let's keep things the way you implemented and revisit if such an issue ever comes up.

Reply:
That's a really good point about protecting the "real" track for apps that have the workflow:
1. Automation uploads to "private" track
2. Release Management promotes to "public" track

This would only apply to Fenix, but would still be valuable. I've updated #88